### PR TITLE
Groups and tests

### PR DESCRIPTION
**Pull request by loosebazooka**

### CODE REVIEW

**Comment by loosebazooka** [comment]
TestProject isn't the most perfect test project class, there's some mixing of a builder and some other stuff, but it serves the purposes of the tests pretty well. I can probably make it a more meaningful structure if we need that.

File: src/test/java/com/google/cloud/tools/gradle/appengine/AppEnginePluginTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            .addAutoDetectingBuildFile()
+            .addAppEngineWebXml()
+            .applyGradleRunner("tasks");
+
+    Assert.assertThat(
+        buildResult.getOutput(),
+        CoreMatchers.containsString(AppEngineStandardPlugin.APP_ENGINE_STANDARD_TASK_GROUP));
+    Assert.assertThat(
+        buildResult.getOutput(),
+        CoreMatchers.not(
```
Comment (akerekes) [inline src/test/java/com/google/cloud/tools/gradle/appengine/AppEnginePluginTest.java]: You could use static imports, and then the assertions would read more naturally: assertThat X not containsString Y.

File: src/test/java/com/google/cloud/tools/gradle/appengine/TestProject.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import org.gradle.api.Project;
+import org.gradle.api.internal.project.ProjectInternal;
+import org.gradle.api.plugins.JavaPlugin;
+import org.gradle.api.plugins.WarPlugin;
+import org.gradle.testfixtures.ProjectBuilder;
+import org.gradle.testkit.runner.BuildResult;
+import org.gradle.testkit.runner.GradleRunner;
+
+/** Create test projects for plugin test runs. */
+public class TestProject {
```
Comment (akerekes) [inline src/test/java/com/google/cloud/tools/gradle/appengine/TestProject.java]: If you defined this class as `org.junit.rules.ExternalResource` you could get rid of a few constructor/method calls creating an instance in test methods.

File: src/main/java/com/google/cloud/tools/gradle/appengine/flexible/AppEngineFlexiblePlugin.java [hunk 3]
```diff
@@ -35,11 +35,10 @@
 /** Plugin definition for App Engine flexible environments. */
 public class AppEngineFlexiblePlugin implements Plugin<Project> {
 
+  public static final String APP_ENGINE_FLEXIBLE_TASK_GROUP = "App Engine flexible environment";
```
Comment (patflynn) [inline src/main/java/com/google/cloud/tools/gradle/appengine/flexible/AppEngineFlexiblePlugin.java]: flexible and standard should be capitalized.

File: src/test/java/com/google/cloud/tools/gradle/appengine/TestProject.java (hunk 2 above)
Comment (loosebazooka) [inline src/test/java/com/google/cloud/tools/gradle/appengine/TestProject.java]: Yeah I wasn't too fond of the fact that you have to mess with TemporaryFolder when using ExternalResource with it. Maybe I'll come back to this another day.

**Pull request merged**